tests/widget.js

```diff
@@ -1210,5 +1210,74 @@ describe( 'Widget', () => {
 
 			expect( getModelData( model ) ).to.equal( '<paragraph>bar</paragraph>[<widget></widget>]<paragraph>foo</paragraph>' );
 		} );
+
+		it( 'should select the most top-outer widget if widgets are nested', () => {
```

Comment:
Missing scenarios:

* multiple widgets (on the same level) in the selection,
* some additional element between the outer and nested widget (widget > container > widget)